File: nom-api/Nom.Orch/Models/Recipe/DishGroupModel.cs
using System.Collections.Generic;

namespace Nom.Orch.Models.Recipe
{
    public class DishGroupModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Slug { get; set; } = string.Empty;
        public int RecipeCount { get; set; }
    }

    /// <summary>One member of a dish group, shaped for the variations rail.</summary>
    public class DishGroupRecipeModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Image { get; set; }
        public decimal? Rating { get; set; }
    }

    public class DishGroupDetailModel : DishGroupModel
    {
        public List<DishGroupRecipeModel> Recipes { get; set; } = new();
    }
}

File: nom-ui/src/app/core/models/dish-group.model.ts
/** A canonical dish group ("chocolate chip cookies") with its visible-member count. */
export interface DishGroupModel {
  id: number;
  name: string;
  slug: string;
  recipeCount: number;
}

/** One member of a dish group, shaped for rails and grids. */
export interface DishGroupRecipeModel {
  id: number;
  name: string;
  image: string | null;
  rating: number | null;
}

export interface DishGroupDetailModel extends DishGroupModel {
  recipes: DishGroupRecipeModel[];
}

/** Compact dish-group reference carried on the recipe detail response. */
export interface RecipeDishGroupRef {
  id: number;
  name: string;
  slug: string;
}
